src/helpers/helpers/fairness_measures.py

```diff
@@ -0,0 +1,138 @@
+    return di
+
+
+def disparate_impact_d(scores, attr):
+    """
+    Computes disparate impact on decision level
+    
+    """
+    a_mask = attr == 1
+    di = np.abs(
+        (scores[~a_mask] >= 0.5).mean() - (scores[a_mask] >= 0.5).mean()
+    )
+    return di
+
+def equal_opportunity_p(scores, attr, labels):
+    """
+    Computes equal opportunity on probability level
+    
+    """
+    a_mask = attr == 1
+    y_mask = labels == 1
+
+    eopp = disparate_impact_p(scores[y_mask], a_mask[y_mask])
+
+    return eopp
+    
+def equal_opportunity_d(scores, attr, labels):
+    """
+    Computes equal opportunity on decision level
+    
+    """
+    a_mask = attr == 1
+    y_mask = labels == 1
+
+    eopp = disparate_impact_d(scores[y_mask], a_mask[y_mask])
+
+    return eopp   
+    
+
+def equalised_odds_p(scores, attr, labels):
+    """
+    Computes equalised odds on probability level
+    
+    """
+    a_mask = attr == 1
+    y_mask = labels == 1
+
+    eo_0 = disparate_impact_p(scores[~y_mask], a_mask[~y_mask])
+    eo_1 = disparate_impact_p(scores[y_mask], a_mask[y_mask])
+    eo = np.mean([eo_0, eo_1])
+
+    return eo
+
+
+def equalised_odds_d(scores, attr, labels):
+    """
+    Computes equalised odds on decision level
+    
+    """
+    a_mask = attr == 1
+    y_mask = labels == 1
+
+    eo_0 = disparate_impact_d(scores[~y_mask], a_mask[~y_mask])
+    eo_1 = disparate_impact_d(scores[y_mask], a_mask[y_mask])
+    eo = np.mean([eo_0, eo_1])
+
+    return eo
+
+
+def calibration(scores, attr, labels, n):
+    """
+    Computes calibration
+    
+    """
+    a_mask = attr == 1
+    y_mask = labels == 1
+
+    # Count differences between two groups within bins
+    bins = np.linspace(0, 1, n + 1)
+
+    fair_1 = 0
+    fair_0 = 0
+    for i in range(len(bins) - 1):
+        fair_1 += 1.0 - np.abs(
+            (
+                (scores[~a_mask & y_mask] > bins[i])
+                & (scores[~a_mask & y_mask] < bins[i + 1])
+            ).sum()
+            / (
+                (scores[~a_mask] > bins[i]) & (scores[~a_mask] < bins[i + 1])
+            ).sum()
+            - (
+                (scores[a_mask & y_mask] > bins[i])
+                & (scores[a_mask & y_mask] < bins[i + 1])
+            ).sum()
+            / (
+                (scores[a_mask] > bins[i]) & (scores[a_mask] < bins[i + 1])
+            ).sum()
+        )
+        fair_0 += 1.0 - np.abs(
+            (
+                (scores[~a_mask & ~y_mask] > bins[i])
+                & (scores[~a_mask & ~y_mask] < bins[i + 1])
+            ).sum()
+            / (
+                (scores[~a_mask] > bins[i]) & (scores[~a_mask] < bins[i + 1])
+            ).sum()
+            - (
+                (scores[a_mask & ~y_mask] > bins[i])
+                & (scores[a_mask & ~y_mask] < bins[i + 1])
+            ).sum()
+            / (
+                (scores[a_mask] > bins[i]) & (scores[a_mask] < bins[i + 1])
+            ).sum()
+        )
```

Comment:
Can we break this up and give each block a descriptive variable name? A bit hard to see what's going on here.

Reply:
Yes, I'll make it more clear.